src/Gui/SceneInspector.cpp

```diff
@@ -82,42 +102,131 @@ void SceneModel::setNode(SoNode* node)
     this->clear();
     this->setHeaderData(0, Qt::Horizontal, tr("Nodes"), Qt::DisplayRole);
 
-    this->insertColumns(0,2);
+    this->insertColumns(0,(int)Column::COUNT);
     this->insertRows(0,1);
     setNode(this->index(0, 0), node);
 }
 
+static std::string_view formatSoSwitchValue(int32_t value)
+{
+    switch (value)
+    {
+    case SO_SWITCH_NONE:
+        return {"None"};
+    case SO_SWITCH_INHERIT:
+        return {"Inherit"};
+    case SO_SWITCH_ALL:
+        return {"All"};
+    default:
+        return {"Child"};
+    }
+}
+
+static std::string_view formatSoSeparatorCacheEnabled(int32_t value)
+{
+    switch (value)
+    {
+    case SoSeparator::OFF:
+        return {"Off"};
+    case SoSeparator::ON:
+        return {"On"};
+    case SoSeparator::AUTO:
+        return {"Auto"};
+    default:
+        throw Base::ValueError();
+    }
+}
+
+static std::string_view formatSoDrawStyleElement(int32_t value)
+{
+    switch (value)
+    {
+    case SoDrawStyleElement::FILLED:
+        return {"Filled"};
+    case SoDrawStyleElement::LINES:
+        return {"Lines"};
+    case SoDrawStyleElement::POINTS:
+        return {"Points"};
+    case SoDrawStyleElement::INVISIBLE:
+        return {"Invisible"};
+    default:
+        throw Base::ValueError();
+    }
+}
+
+static std::string_view formatSoPickStyleElement(int32_t value)
+{
+    switch (value)
+    {
+    case SoPickStyleElement::SHAPE:
+        return {"Shape"};
+    case SoPickStyleElement::BOUNDING_BOX:
+        return {"BoundingBox"};
+    case SoPickStyleElement::UNPICKABLE:
+        return {"Unpickable"};
+    case SoPickStyleElement::SHAPE_ON_TOP:
+        return {"ShapeOnTop"};
+    case SoPickStyleElement::BOUNDING_BOX_ON_TOP:
+        return {"BoundingBoxOnTop"};
+    case SoPickStyleElement::SHAPE_FRONTFACES:
+        return {"ShapeFrontFaces"};
+    default:
+        throw Base::ValueError();
+    }
+}
+
 void SceneModel::setNode(QModelIndex index, SoNode* node)
 {
-    this->setData(index, QVariant(QString::fromLatin1(QByteArray(node->getTypeId().getName()))));
+    this->setData(index.siblingAtColumn((int)Column::INVENTOR_TREE),
+        QVariant(QString::fromLatin1(QByteArray(node->getTypeId().getName()))));
+
+    QHash<SoNode*, QString>::iterator it = nodeNames.find(node);
+    QString name;
+    if (it != nodeNames.end()) {
+        name = it.value();
+    } else {
+        name = QString::fromLatin1(QByteArray(node->getName()));
+    }

```

Comment:
```suggestion
    const QString name {
        (it != nodeNames.end()) ? it.value()
                                : QString::fromLatin1(QByteArray(node->getName())); 
    };
```
btw, can't you use `QString::fromLatin1(node->getName())` without `QByteArray`?

Reply:
I think it's only QString that has the constructor private, QString::fromLatin1 can take a `char *` IIRC

Reply:
my bad, tested it and it doesn't work :S

Reply:
still the suggestion above would be nice